greymatter-io/gitops-core
# OIDC filter toggle

[sc-16859]
This add a toggle and defaults for OIDC configuration within the mesh.

To test implementation:
Go to http://20.84.197.187:10808/, you should see a redirect to keycloak login. Login with username:password of exxon:exxon, and you should be redirected back to the dashboard. You should be able to look at the edge and see that the OIDC Authentication filter is turned on, along with a few others. 

To test cue generation: 
We have modified the `edge.cue` provided in this file to use OIDC Auth on the Edge, with an external Keycloak provider. To generate the configuration for the edge, run `cue eval -c ./gm/outputs --out text -e mesh_configs_yaml` from the root of this repo, and then search for `listener_key: edge`. If you have `yq`, you can instead run `cue eval -c ./gm/outputs --out text -e mesh_configs_yaml | yq -o=json '. | select(.listener_key=="edge") | sort_keys(.)'` and get the edge listener directly. 

If you have access to the azure deployment, you can compare the generated cue against what is currently working in the `exxon-demo` with `greymatter get listener -k edge` and ensure that the json output matches.
